server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeHealth.java

```diff
@@ -0,0 +1,323 @@
+
+/**
+ * Cluster Awareness health information
+ *
+ */
+public class ClusterAwarenessAttributeHealth implements Iterable<ClusterAwarenessAttributeValueHealth>, Writeable, ToXContentFragment {
+
+    private final String awarenessAttributeName;
+    private Map<String, ClusterAwarenessAttributeValueHealth> awarenessAttributeValueHealthMap;
+
+    /**
+     * Creates Awareness AttributeValue Health information
+     *
+     * @param awarenessAttributeValue Awareness Attribute value ie zone, rack etc
+     * @param weightedRoutingWeights  Weights Map of Awareness Attribute
+     * @param displayUnassignedShardLevelInfo Governs if unassigned info should be visible or not
+     * @param shardAllocationPerNode Shard Allocation per node
+     * @param nodeMap The nodeMap stored in cluster state
+     * @param totalShards Total shards currently in cluster
+     */
+    public ClusterAwarenessAttributeHealth(
+        String awarenessAttributeValue,
+        Map<String, Double> weightedRoutingWeights,
+        boolean displayUnassignedShardLevelInfo,
+        Map<String, NodeShardInfo> shardAllocationPerNode,
+        ImmutableOpenMap<String, DiscoveryNode> nodeMap,
+        int totalShards
+    ) {
+
+        awarenessAttributeName = awarenessAttributeValue;
+
+        // This is the Map which is storing the per attribute value health stats.
+        // The key would be the attribute value like rack-1 and the stats would be stored as value.
+        Map<String, ClusterAwarenessAttributeValueHealth> attributesHealth = new HashMap<>();
+
+        // This is the map that would store all the stats per attribute ie
+        // health stats for rack-1, rack-2 etc.
+        awarenessAttributeValueHealthMap = new HashMap<>(Collections.emptyMap());
+        String attributeValue;
+
+        if (!nodeMap.isEmpty()) {
+            Iterator<String> iter = nodeMap.keysIt();
+            while (iter.hasNext()) {
+                ClusterAwarenessAttributeValueHealth clusterAwarenessAttributeValueHealth = null;
+                String node = iter.next();
+                DiscoveryNode nodeDiscovery = nodeMap.get(node);
+                Map<String, String> nodeAttributes = nodeDiscovery.getAttributes();
+                if (!nodeAttributes.isEmpty()) {
+                    if (nodeAttributes.containsKey(awarenessAttributeName)) {
+                        attributeValue = nodeAttributes.get(awarenessAttributeName);
+
+                        if (!attributesHealth.containsKey(attributeValue)) {
+                            clusterAwarenessAttributeValueHealth = new ClusterAwarenessAttributeValueHealth(attributeValue);
+                            clusterAwarenessAttributeValueHealth.setNodes(0);
+                            clusterAwarenessAttributeValueHealth.setActiveShards(0);
+                            clusterAwarenessAttributeValueHealth.setInitializingShards(0);
+                            clusterAwarenessAttributeValueHealth.setRelocatingShards(0);
```

Comment:
no need to se this defaults

Reply:
Did that for readability standpoint will remove